### PR TITLE
Decay eps and dqn static

### PR DESCRIPTION
**Pull request by Szymon-Gut**
I have moved static fields in DQNAgent to instance attributes as we disscussed in #208.
I have also fixed big bug in LearningTask, what I also described in #209 (epsilon was decreasing after each agent's step so It falls to 0 really fast, now It decreases after each episode).

### CODE REVIEW

File: academia/agents/dqn_agent.py [hunk 1]
```diff
@@ -90,39 +98,34 @@ class DQNAgent(EpsilonGreedyAgent):
           from ``torch.nn.Module`` and is appropriate for the task.
         - The agent's exploration-exploitation strategy is based on epsilon-greedy method.
         - The __soft_update_target method updates the target network weights from the main network's weights
-          based on strategy target_weights = TAU * main_weights + (1 - TAU) * target_weights, where TAU << 1.
+          based on strategy target_weights = tau * main_weights + (1 - tau) * target_weights, where tau << 1.
```
Comment (maciejors) [inline academia/agents/dqn_agent.py:101]: you could link it in docs by typing :attr:`tau`

File: academia/agents/dqn_agent.py [hunk 2]
```diff
@@ -215,13 +218,13 @@ def __soft_update_target(self):
         """
         Updates the target network's weights with the main network's weights.
 
-        It uses soft max strategy so target_weights = TAU * main_weights + (1 - TAU) * target_weights, where TAU << 1.
-        Small value of Tau still covers the statement that target values supposed to be fixed to prevent moving target problem
+        It uses soft max strategy so target_weights = tau * main_weights + (1 - tau) * target_weights, where tau << 1.
```
Comment (maciejors) [inline academia/agents/dqn_agent.py:221]: here too

File: academia/curriculum/learning_task.py [hunk 3]
```diff
@@ -303,11 +303,11 @@ def __run_episode(self, agent: Agent, evaluation_mode: bool = False) -> tuple[fl
 
             if not evaluation_mode:
                 agent.update(state, action, reward, new_state, done)
-                agent.update_exploration()
 
             state = new_state
             episode_reward += reward
             steps_count += 1
+        agent.update_exploration()
```
Comment (maciejors) [inline academia/curriculum/learning_task.py:310]: this should be placed inside the same `if` block as before since we don't want to update exploration in evaluation mode

**Review by maciejors — CHANGES_REQUESTED** [review]
got some suggestions

**Review by maciejors — APPROVED** [review]
